### PR TITLE
krustlet polish and functionality, base64 fix

### PR DESCRIPTION
**Pull request by alexeldeib**

### CODE REVIEW

**Comment by alexeldeib** [comment]
some notes:
- this leaves containerd running so we can do container image cleanup. we can follow up with removing containerd/images completely from these nodes.
- capacity on krustlet nodes is hardcoded. we should fix that for the sake of scheduling + autoscaling.
- hostnetwork only afaik.
- krustlet doesn't seem to detect external IPs for the node. should fix that.

File: parts/linux/cloud-init/artifacts/krustlet.service [hunk 1]
```diff
@@ -0,0 +1,17 @@
+[Unit]
+Description=Krustlet
+
+[Service]
+Restart=on-failure
+RestartSec=5s
+EnvironmentFile=/etc/default/kubelet
```
Comment (marwanad) [inline parts/linux/cloud-init/artifacts/krustlet.service:7]: do we need to read any env vars from /etc/default/kubelet?

Comment (marwanad) [reply parts/linux/cloud-init/artifacts/krustlet.service:7]: seems like KUBELET_NODE_LABELS comes from there?

File: parts/linux/cloud-init/nodecustomdata.yml [hunk 2]
```diff
@@ -173,7 +182,7 @@ write_files:
   content: !!binary |
     {{GetVariableProperty "cloudInitData" "kubeletMonitorSystemdService"}}
 
-{{- if NeedsContainerd}}
```
Comment (marwanad) [inline parts/linux/cloud-init/nodecustomdata.yml:176]: curious: is this diff needed?

Comment (marwanad) [reply parts/linux/cloud-init/nodecustomdata.yml:176]: most ignorance on the difference between {{-if}} vs {{if -}. I assume the else clause but just checking for my sanity.

**Review by marwanad — APPROVED** [review]
just questions to clarify but looks good to me

File: parts/linux/cloud-init/artifacts/krustlet.service (hunk 1 above)
Comment (alexeldeib) [reply parts/linux/cloud-init/artifacts/krustlet.service:7]: Yep just node labels for now

File: parts/linux/cloud-init/nodecustomdata.yml (hunk 2 above)
Comment (alexeldeib) [reply parts/linux/cloud-init/nodecustomdata.yml:176]: It mostly helps keep the generated diff cleaner

**Pull request merged**